core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -472,7 +472,7 @@ object StoreUtils {
     } else {
       parameters.remove(DISKSTORE).foreach { v =>
         if (isPersistent) sb.append(s"'$v' ")
-        else throw Utils.analysisException(
+        else if (!isPersistent && !overflow) throw Utils.analysisException(
```

Comment:
Can you please explain this  check  and why is it required.

Reply:
It is a fix for SNAP-2336(PERSISTENCE as none does not work with user-specified diskstore in overflow configuration). Made a change to allow this.